samples/csharp_dotnetcore/82.skills-sso-cloudadapter/README.md

```diff
@@ -0,0 +1,65 @@
+# SSO with Simple Skill Consumer and Skill
+
+Bot Framework v4 skills echo sample.
+
+This bot has been created using [Bot Framework](https://dev.botframework.com), it shows how to create a simple root bot that sends message activities to a skill bot that echoes it back.
+
+## Prerequisites
+
+- [.NET Core SDK](https://dotnet.microsoft.com/download) version 3.1
+
+  ```bash
+  # determine dotnet version
+  dotnet --version
+  ```
+
+## Key concepts in this sample
+
+The solution includes a parent bot ([`rootBot`](RootBot/Bots/RootBot.cs)) and a skill bot ([`skillBot`](SkillBot/Bots/SkillBot.cs)) and shows how the skill bot can accept OAuth credentials from the root bot, without needing to send it's own OAuthPrompt.
+
+This is the general authentication flow :
+
+1. Root bot prompts user to authenticate with an OAuth prompt card.
+2. Authentication succeeds and the user is granted a token.
+3. User performs and action on the skill bot that requires authentication.
+4. The skill bot sends an OAuth prompt card to the root bot.
+5. The root bot intercepts the OAuth prompt card, aware that the user is already authenticated and that the user should authenticate with the skill via SSO.
+6. Instead of showing the OAuth prompt card to the user, the root bot sends a token exchange request invoke activity along with the token to the skill.
+7. The skill's OAuth prompt receives the token exchange request and uses the token from the root bot to continue authenticating.
+
+## To try this sample
+
+- Clone the repository
+
+    ```bash
+    git clone https://github.com/microsoft/botbuilder-samples.git
+    ```
+
+- Create a bot registration in the azure portal for the **SkillBot** and update [SkillBot/appsettings.json](SkillBot/appsettings.json) with the `MicrosoftAppId` and `MicrosoftAppPassword` of the new bot registration.
+- Update the `BotFrameworkSkills` section in [RootBot/appsettings.json](RootBot/appsettings.json) with the app ID for the skill you created in the previous step.
+- Create a bot registration in the azure portal for the **RootBot** and update [RootBot/appsettings.json](RootBot/appsettings.json) with the `MicrosoftAppId` and `MicrosoftAppPassword` of the new bot registration.
+- Add the **RootBot** `MicrosoftAppId` to the `AllowedCallers` list in [SkillBot/appsettings.json](SkillBot/appsettings.json).
+- Setup the 2 Azure AD applications for SSO as per steps given in [SkillBot Azure AD](https://docs.microsoft.com/en-us/azure/bot-service/bot-builder-authentication-sso?view=azure-bot-service-4.0&tabs=csharp%2Ceml#create-the-azure-ad-identity-for-skillbot) and [RootBot Azure AD](https://docs.microsoft.com/en-us/azure/bot-service/bot-builder-authentication-sso?view=azure-bot-service-4.0&tabs=csharp%2Ceml#create-the-azure-ad-identity-for-rootbot). You will end up with 2 Azure AD applications - one for the skill consumer and one for the skill.
```

Comment:
This step links to a page that asks user to create a new app registration with the additional API permissions. So, should the steps above to create **SkillBot** and **RootBot** refer to this link directly, and this step can be removed?

Reply:
Reworded the readme to make it a bit more clear